### PR TITLE
create_session tests and implementation

### PR DESCRIPTION
**Pull request by ekruse09**

### CODE REVIEW

File: supercreative/CreateAccount/delete_account.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+from supercreative.models import User, UserCourseAssignment
+from django.core.exceptions import ObjectDoesNotExist
+
+
+def delete_user(user_id, role):
```
Comment (N7Ironclad) [inline supercreative/CreateAccount/delete_account.py]: Remove role parameter, should be protected by template and/or view.

File: supercreative/CreateAccount/delete_account.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    # Preconditions
+    if role is None or user_id is None:
+        return False
+
+    if role.lower() != "administrator":
```
Comment (N7Ironclad) [inline supercreative/CreateAccount/delete_account.py]: Can remove this check.

File: supercreative/create_sections/section.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    # Check if the SectionID already exists in the Sections table
+    if Section.objects.filter(section_id=section_id).exists():
+        return False
+
+    # Check if the active user role is Administrator
+    if role != "administrator":
+        return False
+
+    # Check if Section Type is valid: "Lecture", "Lab", or "Discussion"
```
Comment (N7Ironclad) [inline supercreative/create_sections/section.py:26]: Would prefer to use all uppercase and section_type.upper() for roles.

File: supercreative/create_sections/section.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    valid_section_types = ["lecture", "lab", "discussion"]
+
+    if section_type is None:
+        return False
+
```
Comment (N7Ironclad) [inline supercreative/create_sections/section.py:28]: Can combine these two checks into one with an or statement.

File: supercreative/create_sections/section.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
+from supercreative.models import Course, Section
+from django.core.exceptions import ObjectDoesNotExist
+
+
+def create_section(section_id, course, section_type, role):
```
Comment (N7Ironclad) [inline supercreative/create_sections/section.py]: Can remove role parameter, will be protected by template and/or view.

File: supercreative/create_sections/section.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        Course.objects.get(course_id=course.course_id)
+    except ObjectDoesNotExist:
+        return False
+
+    # Check if the SectionID already exists in the Sections table
+    if Section.objects.filter(section_id=section_id).exists():
+        return False
+
+    # Check if the active user role is Administrator
+    if role != "administrator":
```
Comment (N7Ironclad) [inline supercreative/create_sections/section.py]: Can remove this check.

**Review by N7Ironclad — CHANGES_REQUESTED** [review]
Looks good! Just remove role as a parameter and the corresponding checks for it. Role will be checked for in the template and/or view. I will update method descriptions to reflect that in the Design Document.

**Review by N7Ironclad — APPROVED** [review]

**Review by brogers32 — APPROVED** [review]

**Pull request merged**